src/plugins/intel_gpu/tests/transformations/reduce_modification_for_false_keepdims_test.cpp

```diff
@@ -0,0 +1,173 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include <memory>
+#include <ngraph/function.hpp>
+#include <ngraph/opsets/opset10.hpp>
+#include <ngraph/pass/manager.hpp>
+#include <plugin/transformations/reduce_modification_for_false_keepdims.hpp>
+#include <string>
+#include <transformations/init_node_info.hpp>
+#include <transformations/utils/utils.hpp>
+#include <tuple>
+
+#include "test_utils.h"
+#include "common_test_utils/ngraph_test_utils.hpp"
+
+using namespace testing;
+using namespace ::tests;
+
+using InputShape = ngraph::PartialShape;
+using KeepDims = bool;
+using ReduceAxes = std::vector<int64_t>;
+using ReduceType = std::shared_ptr<ngraph::Node>;
+using ReshapeShape = std::vector<size_t>;
+using NeedReshape = bool;
+
+class ReduceModificationTests
+    : public ::testing::Test,
+      public testing::WithParamInterface<
+          std::tuple<ReduceType, InputShape, ReduceAxes, KeepDims, NeedReshape, ReshapeShape>> {
+public:
+    std::shared_ptr<ngraph::Function> f, f_ref;
+
+    void SetUp() override {
+        const auto& reduce_type = std::get<0>(GetParam());
+        const auto& input_shape = std::get<1>(GetParam());
+        const auto& axes = std::get<2>(GetParam());
+        const auto& keep_dims = std::get<3>(GetParam());
+        const auto& need_reshape = std::get<4>(GetParam());
+        const auto& reshape_shape = std::get<5>(GetParam());
+
+        f = get_initial_function(input_shape, axes, reduce_type, keep_dims, need_reshape, reshape_shape);
+        f_ref = get_reference_function(input_shape, axes, reduce_type);
+    }
+
+    static std::shared_ptr<ngraph::Function> get_reference_function(const ngraph::PartialShape& input_shape,
+                                                                    const std::vector<int64_t>& axes,
+                                                                    const ReduceType& reduce_type) {
+        auto input = std::make_shared<ngraph::opset10::Parameter>(ngraph::element::f32, input_shape);
+        auto axes_const = ngraph::opset10::Constant::create(ngraph::element::i64, ngraph::Shape{axes.size()}, axes);
+
+        auto reduce = std::make_shared<ngraph::opset10::ReduceMax>(input, axes_const, true);
+
+        return std::make_shared<ngraph::Function>(ngraph::NodeVector{reduce}, ngraph::ParameterVector{input});
+    }
+
+    static std::shared_ptr<ngraph::Function> get_initial_function(const ngraph::PartialShape& input_shape,
+                                                                  const std::vector<int64_t>& axes,
+                                                                  const ReduceType& reduce,
+                                                                  const bool keep_dim,
+                                                                  const bool need_reshape,
```

Comment:
`need_reshape` is always `!keep_dims`? If so, why do we need to have two separate variables?

Reply:
usage is changed. They all used with difference meaning.

Reply:
relevant tests are added 
[----------] 6 tests from ReduceDecomposeForFalseKeepdimsNotCase/ReduceDecomposeTests
[ RUN      ] ReduceDecomposeForFalseKeepdimsNotCase/ReduceDecomposeTests.CompareFunctions/0
[       OK ] ReduceDecomposeForFalseKeepdimsNotCase/ReduceDecomposeTests.CompareFunctions/0 (0 ms)
[ RUN      ] ReduceDecomposeForFalseKeepdimsNotCase/ReduceDecomposeTests.CompareFunctions/1
[       OK ] ReduceDecomposeForFalseKeepdimsNotCase/ReduceDecomposeTests.CompareFunctions/1 (0 ms)